nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -9,490 +9,492 @@
+    # 4D mask
+    wrong_dim_mask_img = img_4D
 
-    mask_4d_img = nibabel.Nifti1Image(np.ones(shape + (2,)), affine)
+    with pytest.raises(DimensionError, match=_3D_EXPECTED_ERROR_MSG):
+        img_to_signals_labels(
+            imgs=img_4D, labels_img=img_3D, mask_img=wrong_dim_mask_img
+        )
 
-    # labels
-    labels_data = np.zeros(shape, dtype="int32")
-    h0 = shape[0] // 2
-    h1 = shape[1] // 2
-    h2 = shape[2] // 2
-    labels_data[:h0, :h1, :h2] = 1
-    labels_data[:h0, :h1, h2:] = 2
-    labels_data[:h0, h1:, :h2] = 3
-    labels_data[:h0, h1:, h2:] = 4
-    labels_data[h0:, :h1, :h2] = 5
-    labels_data[h0:, :h1, h2:] = 6
-    labels_data[h0:, h1:, :h2] = 7
-    labels_data[h0:, h1:, h2:] = 8
+    with pytest.raises(DimensionError, match=_3D_EXPECTED_ERROR_MSG):
+        signals_to_img_labels(
+            signals=img_4D, labels_img=img_3D, mask_img=wrong_dim_mask_img
+        )
 
-    labels_img = nibabel.Nifti1Image(labels_data, affine)
+    with pytest.raises(DimensionError, match=_3D_EXPECTED_ERROR_MSG):
+        img_to_signals_maps(
+            imgs=img_4D, maps_img=img_4D, mask_img=wrong_dim_mask_img
+        )
 
-    labels_4d_data = np.zeros((shape) + (2,))
-    labels_4d_data[..., 0] = labels_data
-    labels_4d_data[..., 1] = labels_data
-    labels_4d_img = nibabel.Nifti1Image(labels_4d_data, np.eye(4))
 
-    # Without mask
-    # from labels
-    data_img = signal_extraction.signals_to_img_labels(signals, labels_img)
-    data = get_data(data_img)
-    assert data_img.shape == (shape + (n_instants,))
-    assert np.all(data.std(axis=-1) > 0)
+"""errors img_to_signals_labels"""
+
+
+@pytest.mark.parametrize(
+    "shape, affine, error_msg",
+    [
+        ((8, 9, 11), AFFINE, _TEST_SHAPE_ERROR_MSG),
+        (SHAPE, 2 * AFFINE, _TEST_AFFINE_ERROR_MSG),
+    ],
+)
+def test_img_to_signals_labels_bad_inputs(
+    img_4D, img_3D, shape, affine, error_msg
+):
+    bad_labels_img = nibabel.Nifti1Image(np.zeros(shape), affine)
 
-    # verify that 4D label images are refused
-    with pytest.raises(DimensionError, match=_TEST_DIM_ERROR_MSG):
-        signal_extraction.img_to_signals_labels(data_img, labels_4d_img)
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(imgs=img_4D, labels_img=bad_labels_img)
 
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(
+            imgs=img_4D, labels_img=bad_labels_img, mask_img=img_3D
+        )
+
+    bad_mask_img = nibabel.Nifti1Image(np.zeros(shape), affine)
+
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(
+            imgs=img_4D, labels_img=img_3D, mask_img=bad_mask_img
+        )
+
+
+def test_img_to_signals_labels_error_strategy(img_4D, img_3D):
+    with pytest.raises(ValueError, match="Invalid strategy"):
+        img_to_signals_labels(imgs=img_4D, labels_img=img_3D, strategy="foo")
+
+
+"""errors signals_to_img_labels"""
```

Comment:
Same here. I would remove these "global comments". I we feel like we need those to organize the file, then it probably means that we should split it in multiple files.